client/layout/masterbar/logged-in.jsx

```diff
@@ -155,12 +159,20 @@ class MasterbarLoggedIn extends React.Component {
 	}
 
 	render() {
-		const { domainOnlySite, translate, isCheckout, isMigrationInProgress } = this.props;
+		const { domainOnlySite, translate, isCheckout, isMigrationInProgress, siteSlug } = this.props;
 
 		if ( isCheckout === true ) {
 			return (
 				<Masterbar>
 					<div className="masterbar__secure-checkout">
+						<Item
+							url={ '/plans/' + siteSlug }
```

Comment:
Probably not necessary just yet but when we need to, is there a way we can pass a url, maybe through the URL, so this can be controlled externally?

Reply:
Shouldn't be a problem to use a redirect parameter, but it'd be good to know a specific use case first. I could make it smarter by redirecting to the previous screen _except_ in the breaking cases (like Signup and launch flow).

Reply:
The use case that comes to mind is when you click on an upgrade banner, rather than going back to the plans, we take you back to where you clicked the banner. While working on the premium content block, I noticed we use a redirect url parameter already but as you suggested, we can likely just make it smart and go back where you came from (except in some cases).